internal/trace/listener.go

```diff
@@ -118,8 +113,9 @@ func startFunctionExecutionSpan(ctx context.Context, mergeXrayTraces bool) trace
 		tracer.Tag("resource_names", lambdacontext.FunctionName),
 	)
 
-	if traceSource == fromXray && mergeXrayTraces {
-		span.SetTag("_dd.parent_source", traceSource)
+	if parentSpanContext != nil && mergeXrayTraces {
```

Comment:
Is this always true? parentSpanContext isn't set when we receive incomming dd-trace headers?

Reply:
Previously, if we received incoming dd-trace headers, this block would not execute, and the function execution span would not be dropped by the Forwarder. This created a situation where we would have two function execution spans: the one from dd-trace and the one from X-Ray, but only if the function had incoming dd-trace headers. If the function originated the Datadog trace context, it would have one function execution span.

My goal here was to make it so that the dd-trace function execution span is always dropped if merge X-Ray traces is turned on, assuming there was a valid parent. If merge X-Ray traces is turned on, we can assume that X-Ray is turned on that we do not need a second function execution span.

Does this match your understanding of the expected behavior?